network-area-diagram-viewer/src/main/resources/svg.js

```diff
@@ -0,0 +1,448 @@
+            if (idParent) {
+                id = idParent;
+            }
+            updateOnClient(id, position, translation);
+        }
+    }
+
+    function updateOnClient(id, position, translation) {
+        console.log("updateOnClient, moved node id = " + id);
+        translateText(id, translation);
+        for (var edge of svg.getElementsByTagName("nad:edge")) {
+            var node1 = edge.getAttribute("node1");
+            var node2 = edge.getAttribute("node2");
+            console.log("  consider edge (" + node1 + ", " + node2 + ") after moved node " + id);
+            if (node1 === id || node2 === id) {
+                updateEdge(edge, node1, node2, id, position, translation);
+            }
+        }
+    }
+
+    function updateEdge(edgeMetadata, node1, node2, movedNodeId, position, translation) {
+        const DEBUG_EXPERIMENTS = false;
+        if (DEBUG_EXPERIMENTS) {
+            updateEdgeExperiments(edgeMetadata, node1, node2, id, position, translation);
+        } else {
+            updateEdgeUsingRotation(edgeMetadata, node1, node2, id, position, translation);
+        }
+    }
+
+    function updateEdgeUsingRotation(edgeMetadata, node1, node2, movedNodeId, position, translation) {
+        // This edge is adjacent to the moved node
+        console.log("  adjacent edge (" + node1 + ", " + node2 + ")");
+        var edgeId = edgeMetadata.getAttribute("svgid");
+        var edgeSvg = svg.getElementById(edgeId);
+        if (!edgeSvg) {
+            return;
+        }
+
+        // Translate all edge-related graphical elements to the new position
+        translate(svg, edgeSvg, translation);
+
+        // If the other node of the edge is visible in the diagram,
+        // rotate the edge towards it.
+        // rotation centered on moved node
+        // (now we may not "reach" the other node, but we point in the right direction)
+        var otherNodeId = movedNodeId === node1 ? node2 : node1;
+        var otherNodeSvg = svg.getElementById(otherNodeId);
+        console.log("  other node is " + otherNodeId + ", svg = " + otherNodeSvg);
+        if (otherNodeSvg) {
+            rotateEdge(edgeId, edgeSvg, movedNodeId, otherNodeId, otherNodeSvg, position, translation);
+        }
+    }
+
+    function rotateEdge(edgeId, edgeSvg, movedNodeId, otherNodeId, otherNodeSvg, position, translation) {
+        var otherNodeCenter = center(otherNodeSvg);
+        var movedNodeMetadata = svg.querySelectorAll('nad\\\\:node[svgid="' + movedNodeId + '"]')[0];
+        var otherNodeMetadata = svg.querySelectorAll('nad\\\\:node[svgid="' + otherNodeId + '"]')[0];
+        var position0 = {x: movedNodeMetadata.getAttribute("x"), y: movedNodeMetadata.getAttribute("y")};
+        var otherNodePosition0 = {x: otherNodeMetadata.getAttribute("x"), y: otherNodeMetadata.getAttribute("y")};
+        var rotation0 = calcRotation(position0, otherNodePosition0);
+        var rotation1 = calcRotation(position, otherNodeCenter);
```

Comment:
you should not use position (which is the mouse position) to calculate the rotation but position1 = position0 + translation

Reply:
I am not sure to understand. `position` is the current mouse position expressed in the SVG coordinate system, is the exact location where we want to move the dragged element, `position0 + translation` should be equal to `position`, as `translation` has been calculated from mouse position minus the initial `offset` (the initial translation).

I used explicit new `position` and `translation` as two parameters because for some of the experiments/updates it was more convenient to use one or another.

I've changed the name of the `position` parameter in this function to `position1`, and removed the `translation` parameter that was not used.

Previously, I also added and used the "debug" elements to check the calculated rotations.

Reply:
with position the angle is not properly calculated, and you putting the node back to its original position doesn't lead to the same edge positions if you don't drag the node by clicking exactly on its center. In the following example I moved the node one pixel to the right then one pixel back to the left:
![screenshot](https://user-images.githubusercontent.com/66690739/204507493-a792cd78-ddd6-4489-bd1b-e02d013ad640.png)

Reply:
Sorry for the confusion, I now see the issue that you were pointing out.